src/modules/sync/validation/entitySchema.ts

```diff
@@ -0,0 +1,164 @@
+  name: z.string(),
+  folders: z.lazy(() => AssetFolderSchema.array()),
+  codename: z.string(),
+});
+
+export const TaxonomySchema: z.ZodType<TaxonomySyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  terms: z.lazy(() => TaxonomySchema.array()),
+});
+
+export const SnippetSchema: z.ZodType<ContentTypeSnippetsSyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  elements: z.array(SnippetElementsSchemasUnion),
+});
+
+const ContentGroupSchema = z.strictObject({ codename: z.string(), name: z.string() });
+
+export const TypeSchema: z.ZodType<
+  ContentTypeSyncModel,
+  z.ZodTypeDef,
+  { content_groups: ReadonlyArray<Replace<ContentTypeContracts.IContentTypeGroup, { codename: string }>> }
+> = z
+  .strictObject({ content_groups: z.array(ContentGroupSchema) })
+  .passthrough()
+  .transform(obj => ({ ...obj, groups_number: obj.content_groups.length === 0 ? "zero" : "multiple" }))
+  .pipe(
+    z.discriminatedUnion("groups_number", [
+      z.strictObject({
+        name: z.string(),
+        codename: z.string(),
+        groups_number: z.literal("zero"),
+        content_groups: z.array(ContentGroupSchema).length(0),
+        elements: z.array(TypeElementsSchemasUnion),
+      }),
+      z.strictObject({
+        name: z.string(),
+        codename: z.string(),
+        groups_number: z.literal("multiple"),
+        content_groups: z.array(ContentGroupSchema).min(1),
+        elements: z.array(TypeElementWithGroupSchemasUnion),
+      }),
+    ])
+      .refine(a =>
+        a.groups_number === "zero"
+          ? true
+          : a.elements.every(e => a.content_groups.map(c => c.codename).includes(e.content_group.codename)), {
+        message: "Content group codename must be one of the type's content group codename",
+      }),
+  )
+  .transform(obj => omit(obj, ["groups_number"]));
+
+export const WebSpotlightSchema: z.ZodType<WebSpotlightSyncModel> = z.strictObject({
+  enabled: z.boolean(),
+  root_type: CodenameReferenceSchema.nullable(),
+});
+
+export const CollectionSchema: z.ZodType<CollectionSyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+});
+
+export const SpaceSchema: z.ZodType<SpaceSyncModel> = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  web_spotlight_root_item: CodenameReferenceSchema.optional(),
+  collections: z.array(CodenameReferenceSchema),
+});
+
+export const LanguageSchema: z.ZodType<LanguageSyncModel> = z.discriminatedUnion("is_default", [
+  z.strictObject(
+    {
+      name: z.string(),
+      codename: z.string(),
+      is_active: z.boolean(),
+      is_default: z.literal(true),
+    },
+  ),
+  z.strictObject(
+    {
+      name: z.string(),
+      codename: z.string(),
+      is_active: z.boolean(),
+      is_default: z.literal(false),
+      fallback_language: CodenameReferenceSchema,
+    },
+  ),
+]);
+
+const WorkflowColorSchema = z.union([
```

Comment:
Don't forget to use explicit types for the schemas as it is easy to make a mistake here, especially in this union. If you forget a a color, the TS won't complain even when you use an explicit type for the constant. The best solution is probably to have an explicitly typed object with keys being the colors and values being the appropriate schema and then generate the union from the object with `Object.values`. This way TS would check that you don't forget a color.

Reply:
What is the difference between `z.enum` and `z.union` filled with `z.literal`. I noticed you used both in different places. Can we use the `z.enum` here as well? It seems to be simpler.